### PR TITLE
Validate and encode strings as UTF-8

### LINKED ISSUE #6426: Match Bug - Unable to find entity by attribute
Hi,

I've deployed on GKE typedb server (version 2.3.2). I am connecting with the console (version 2.1.3) to the remote db. 

The DB schema has defined the entity `sentence`, that has the attribute `text`. 

With the query `match $s isa sentence, has text $t; get $s, $t;` I obtain the result:
```
...
{ $s iid 0x966e8008800000000000000d isa sentence; $t "Lo sportello metallico automatico a scorrimento verticale protegge l’operatore durante le operazioni di prelievo e deposito in baia durante la movimentazione dell’elevatore all'interno della macchina. Durante la fase di apertura/chiusura dello sportello, è attiva la barriera di protezione fotoelettrica multi raggio.  Tutte le baie di accesso sono dotate di sportello a chiusura automatica. L’apertura e la chiusura dello sportello a chiusura automatica di baia in modalità manuale è comandata dai pulsanti ausiliari montati lateralmente a ciascuna baia in prossimità del pulsante di arresto di emergenza a fungo." isa text; }
...
```

but, If I run the query:
```
match $x isa sentence, has text "Lo sportello metallico automatico a scorrimento verticale protegge l’operatore durante le operazioni di prelievo e deposito in baia durante la movimentazione dell’elevatore all'interno della macchina. Durante la fase di apertura/chiusura dello sportello, è attiva la barriera di protezione fotoelettrica multi raggio.  Tutte le baie di accesso sono dotate di sportello a chiusura automatica. L’apertura e la chiusura dello sportello a chiusura automatica di baia in modalità manuale è comandata dai pulsanti ausiliari montati lateralmente a ciascuna baia in prossimità del pulsante di arresto di emergenza a fungo."; get $x;
```

i receive back:
```
answers: 0, duration: 76 ms
```

I have a local db (version 2.2.0) with the same data. By running the same query I instead receive the expected entity.

Do you have any idea about how to solve this issue without downgrading the docker container to typedb version 2.2.0?

### LINKED ISSUE #6435: Can not show chinese or japanese correctly on Windows
## Description

After inserting chinese or japanese data into typedb both with console and PythonAPI, the result of match statement is displayed as “?�O”.

## Environment

1. OS (where TypeDB server runs): Windows 10
2. TypeDB version (and platform): TypeDB 2.3.2
3. TypeDB client: client-python, console
4. Other environment details:

## Reproducible Steps

Steps to create the smallest reproducible scenario:
1. Run 'transaction testdb data write'
2. Run 'insert $t isa person, has name “张三”;'
3. See error '$_0 “??O” isa name; $t iid 0x966e80017fffffffffffffff isa person;'

## Expected Output

'$_0 “张三” isa name; $t iid 0x966e80017fffffffffffffff isa person;'

## Actual Output

'$_0 “??O” isa name; $t iid 0x966e80017fffffffffffffff isa person;'
 
## Additional information

It is the same problem also on Typedb Workbase.
564e68589350226eff1b83d8d6537c8b5ea69679

### PR DESCRIPTION
**Pull request by flyingsilverfin**
## What is the goal of this PR?

We use UTF8 encoding when converting a String to byte arrays. We also test that all unicode characters are correctly encoded and decoded by the database, throwing invalid strings (ie. not encodable) as an exception.

## What are the changes implemented in this PR?

* The conversion from Strings to ByteArray now uses the provided (UTF8) encoding
* This changes requires everyone reload their data. We fix #6426. This may also fix #6435.
* Test all unicode strings are encodable
* Throw if provided an invalid string

### CODE REVIEW

File: common/exception/ErrorMessage.java [hunk 1]
```diff
@@ -278,64 +278,66 @@ public static void loadConstants() {
                 new ThingWrite(1, "Attempted an illegal write of a new '%s' of abstract type '%s'.");
         public static final ThingWrite ILLEGAL_STRING_SIZE =
                 new ThingWrite(2, "Attempted to insert a string larger than the maximum possible size: %s bytes.");
+        public static final ThingWrite UNENCODABLE_STRING =
```
Comment (flyingsilverfin) [inline common/exception/ErrorMessage.java:281]: simplify without impl about bytes

**Review by lolski — APPROVED** [review]
